tasks/41_sort-by-bits/sort-by-bits.js

```diff
@@ -1,18 +1,27 @@
 export const sortByBits = (arr) => {
-  arr.forEach((cur, index) => {
-    let nextNumber = Math.abs(cur);
-    let binaryFormat = cur >= 0 ? [] : [1];
-
-    while (nextNumber > 0) {
-      if (nextNumber % 2 === 1) {
-        binaryFormat.push(1);
-      }
-      nextNumber = Math.floor(nextNumber / 2);
+  let arrSort = arr.sort((a, b) => {
+    let aOne = a.toString(2).replace(/[0]/g, '').length;
+    let bOne = b.toString(2).replace(/[0]/g, '').length;
+    if (aOne > bOne) {
+      return 1;
+    } else if (aOne < bOne) {
+      return -1;
+    } else {
+      return a - b;
     }
-    arr[index] = {value: cur, length: binaryFormat.length};
   });
-  arr.sort((a, b) => a.length - b.length || a.value - b.value)
-      .forEach((cur, index) => arr[index] = cur.value);
-  return arr;
+  return arrSort;
+};
+
+
+/* Second solution:
+
+ export const sortByBits = (arr) => {
+  let arrOfOne = arr.map((cur) => {
+    return {val: cur, length: cur.toString(2).replace(/[0]/g, '')};
+  });
+  return arrOfOne.sort((a, b) => (a.length - b.length || a.val - b.val))
+      .map((cur) => cur.val);
```

Comment:
bad thing about this solution that you have 3 loops instead of 1
good thing that you are doing changes step by step